clippy_lints/src/utils/internal_lints/lint_without_lint_pass.rs

```diff
@@ -175,25 +166,15 @@ impl<'tcx> LateLintPass<'tcx> for LintWithoutLintPass {
                 }) = field.expr.kind
                 {
                     let sym_str = sym.as_str();
-                    if is_lint_ref_ty {
-                        if sym_str == "default lint description" {
-                            span_lint(
-                                cx,
-                                DEFAULT_LINT,
-                                item.span,
-                                format!("the lint `{}` has the default lint description", item.ident.name),
-                            );
-                        }
-
-                        self.declared_lints.insert(item.ident.name, item.span);
-                    } else if sym_str == "default deprecation note" {
+                    if sym_str == "default lint description" {
                         span_lint(
                             cx,
-                            DEFAULT_DEPRECATION_REASON,
```

Comment:
The `DEFAULT_DEPRECATION_REASON` is now no longer emitted. Do we still want to keep this lint, as the `update_lints` file can still generate the default reason?

Alternatively we can make the `--reason` flag mandatory for the deprecation command.

Reply:
Just made the reason required given that you just end up getting an error anyways.